internal/service/timestreaminfluxdb/db_instance.go

```diff
@@ -0,0 +1,1182 @@
+				},
+				Description: `The amount of storage to allocate for your DB storage type in GiB (gibibytes). 
+					This argument is required. This argument has a minimum value of 
+					20 and a maximum value of 16384`,
+			},
+			"arn": framework.ARNAttributeComputedOnly(),
+			"availability_zone": schema.StringAttribute{
+				Computed: true,
+			},
+			"bucket": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(2),
+					stringvalidator.LengthAtMost(64),
+					stringvalidator.RegexMatches(
+						// Taken from the model for TimestreamInfluxDB in AWS SDK Go V2
+						// https://github.com/aws/aws-sdk-go-v2/blob/8209abb7fa1aeb513228b4d8c1a459aeb6209d4d/codegen/sdk-codegen/aws-models/timestream-influxdb.json#L768
+						regexache.MustCompile("^[^_][^\"]*$"),
+						"",
+					),
+				},
+				Description: `The name of the initial InfluxDB bucket. All InfluxDB data is stored in a bucket. 
+					A bucket combines the concept of a database and a retention period (the duration of time 
+					that each data point persists). A bucket belongs to an organization. This argument is optional. 
+					If not provided, defaults to "bucket"`,
+			},
+			"db_instance_type": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						"db.influx.medium",
+						"db.influx.large",
+						"db.influx.xlarge",
+						"db.influx.2xlarge",
+						"db.influx.4xlarge",
+						"db.influx.8xlarge",
+						"db.influx.12xlarge",
+						"db.influx.16xlarge",
+					),
+				},
+				Description: `The Timestream for InfluxDB DB instance type to run InfluxDB on. 
+					This argument is required. Possible values: 
+					"db.influx.medium", 
+					"db.influx.large", 
+					"db.influx.xlarge", 
+					"db.influx.2xlarge", 
+					"db.influx.4xlarge", 
+					"db.influx.8xlarge", 
+					"db.influx.12xlarge", 
+					"db.influx.16xlarge"`,
+			},
+			"db_parameter_group_identifier": schema.StringAttribute{
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(3),
+					stringvalidator.LengthAtMost(64),
+					stringvalidator.RegexMatches(
+						// Taken from the model for TimestreamInfluxDB in AWS SDK Go V2
+						// https://github.com/aws/aws-sdk-go-v2/blob/8209abb7fa1aeb513228b4d8c1a459aeb6209d4d/codegen/sdk-codegen/aws-models/timestream-influxdb.json#L1390
+						regexache.MustCompile("^[a-zA-Z0-9]+$"),
+						"",
+					),
+				},
+			},
+			"db_storage_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						"InfluxIOIncludedT1",
```

Comment:
Rather than using indexes with the names, would it be better to use the IOPS in the naming. e.g. `InfluxDBIOIncluded3K`

Reply:
This is the value provided to the AWS SDK Go. I want to be consistent with the AWS SDK Go.

My latest PR for `Update` changes all of these values from hardcoded strings to instead directly use `awstypes.DbStorageType`. In the AWS SDK Go they are defined as
```
// Enum values for DbStorageType
const (
	DbStorageTypeInfluxIoIncludedT1 DbStorageType = "InfluxIOIncludedT1"
	DbStorageTypeInfluxIoIncludedT2 DbStorageType = "InfluxIOIncludedT2"
	DbStorageTypeInfluxIoIncludedT3 DbStorageType = "InfluxIOIncludedT3"
)

```